File: Sources/Gantry/App/BuildVariant.swift
import Foundation

/// Which edition of Gantry this binary is. LITE is the same codebase built with `-DGANTRY_LITE`
/// (`scripts/build-app.sh lite`): the tray/menu-bar surface with printers, notifications and a short
/// settings sheet, and nothing else. Everything the full app adds on top — Spoolbase, the detail
/// view, maintenance, automations, diagnostics, fleet statistics, Telegram, the LAN web dashboard,
/// the floating window and the edge dock — is switched off at compile time here rather than being
/// deleted, so both editions keep building from one branch.
///
/// `isLite` is a compile-time constant, so the branches it guards fold away and the LITE build cannot
/// reach a surface it does not ship.
enum Build {
#if GANTRY_LITE
    static let isLite = true
#else
    static let isLite = false
#endif

    /// Product name shown in menus, alerts and the About row. The bundle carries the same string
    /// (build-app.sh sets CFBundleName/CFBundleDisplayName), so both stay in step.
    static var appName: String { isLite ? "Gantry LITE" : "Gantry" }

    /// True when the full edition ships the feature; LITE keeps only the tray essentials.
    static var hasExtras: Bool { !isLite }
}

File: windows/Gantry.Windows/Services/BuildVariant.cs
namespace Gantry.Services;

/// <summary>
/// Which edition this build is. LITE is the same sources compiled with the GANTRY_LITE constant
/// (<c>dotnet build -p:GantryEdition=lite</c>): tray icon, printers, notifications and a short
/// settings window, and nothing else. Spoolbase, the detail window, maintenance, automations,
/// diagnostics, fleet statistics, Telegram, the LAN web dashboard, the floating window and the edge
/// dock are switched off here instead of being deleted, so both editions build from one branch.
/// </summary>
public static class Build
{
    /// <remarks>A read-only property rather than a const: a const would make every
    /// <c>if (Build.HasExtras)</c> body unreachable code and drown the build in CS0162.</remarks>
    public static bool IsLite { get; } =
#if GANTRY_LITE
        true;
#else
        false;
#endif

    /// <summary>True when this build ships the full feature set.</summary>
    public static bool HasExtras => !IsLite;

    /// <summary>Product name for menus, tooltips and the About row; matches the installed app name.</summary>
    public static string AppName => IsLite ? "Gantry LITE" : "Gantry";
}
